### PR TITLE
Upgrade to react-native 0.71

### PR DESCRIPTION
**Pull request by elbywan**
> ~**⚠️ This PR is blocked until we have XCode14 installed on our own hosted runners.**~

#### This PR upgrades the `react-native` LLM dependency to `v0.71.6`.

As you can imagine there as **many** side effects, most notably:

- Dropping iOS 12 support
- Cleaning up the deps:  some unused polyfills and incompatible libs were removed  and we now use better alternatives
- Massive package upgrades:
  - _react_ : `17` -> `18`
  - _gradle_: `4` -> `7`
  - _metro_: `0.67` -> `0.76`
  - _expo_: `43` -> `48`
  - _and a lot of other minor package upgrades…_

### ❓ Context

- **Impacted projects**: `LLM` `@ledgerhq/native-ui` 
- **Linked resource(s)**: [confluence](https://ledgerhq.atlassian.net/wiki/spaces/WALLETCO/pages/4015030324/LLM+Upgrade+to+React+Native+0.71) 

### ✅ Checklist

- [x] **Test coverage:** CI builds/e2e should at least prevent obvious failures - full QA pass will still be needed
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

N/A

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/package.json
Comment (haammar-ledger) [inline apps/ledger-live-desktop/package.json:1]: Don't we want to upgrade LLD to react 18 too, to keep it aligned with LLM?

File: libs/ledger-live-common/package.json
Comment (haammar-ledger) [inline libs/ledger-live-common/package.json:1]: (same question for LLC, and others using react 17)

File: apps/ledger-live-desktop/package.json
Comment (elbywan) [reply apps/ledger-live-desktop/package.json:1]: Ultimately maybe, but it's not in the scope of this PR and it involves unforeseen consequences.

File: libs/ledger-live-common/package.json
Comment (elbywan) [reply libs/ledger-live-common/package.json:1]: In the long run, yes, definitely.

**Review by valpinkman — APPROVED** [review]

**Pull request merged**